example/spec/helpers/granuleUtils.js

```diff
@@ -2,10 +2,53 @@
 
 const fs = require('fs-extra');
 const {
-  aws: { s3 },
+  aws: { buildS3Uri, parseS3Uri, s3 },
   stringUtils: { globalReplace },
   testUtils: { randomStringFromRegex }
 } = require('@cumulus/common');
+const path = require('path');
+const cloneDeep = require('lodash.clonedeep');
+
+/**
+ * Adds updated url_path to a granule files object
+ *
+ * @param  {Array<object>} files - array of granule files
+ * @param  {string} testId - Test ID to insert into url_path per-granule
+ * @param  {string} collectionUrlPath - collection
+ */
+function addUrlPathToGranuleFiles(files, testId, collectionUrlPath) {
+  const updatedFiles = cloneDeep(files);
+  return updatedFiles.map((file) => {
+    const fileUpdate = cloneDeep(file);
+    const updatedUrlPath = Object.is(file.url_path, undefined) ? collectionUrlPath : `${file.url_path}/`;
```

Comment:
Does `collectionUrlPath` always end with a trailing slash? Or do we need to add some safety for both cases?

Reply:
For the purposes of the existing tests, yes.      It depends on the collection fixture.